extensions/exposition/components/octets.storage/operations/store.js

```diff
@@ -1,11 +1,113 @@
 'use strict'
 
-function store (input, context) {
-  const { storage, request, accept, meta } = input
+const { Readable } = require('node:stream')
+const { Err } = require('error-value')
+const { match } = require('matchacho')
+
+async function store (input, context) {
+  const { storage, request, accept, trust } = input
   const path = request.url
   const claim = request.headers['content-type']
+  const meta = parseMeta(request.headers['content-meta'])
+  const body = await download(request, trust)
+
+  if (body instanceof Error)
+    return body
+
+  return context.storages[storage].put(path, body, { claim, accept, meta })
+}
+
+/**
+ * @param {string | string[] | undefined} value
+ * @returns {Record<string, string>}
+ */
+function parseMeta (value) {
+  const meta = {}
+
+  if (value === undefined)
+    return meta
+
+  if (Array.isArray(value))
+    value = value.join(',')
+
+  for (const pair of value.split(',')) {
```

Comment:
NIT: in case of array you first join the array to string just to split it on the next line.

Can we do the opposite?

```js
if (typeof value === 'string')
  value = value.split(',')

for (const pair of value) {
```